libevm/stateconf/conf.go

```diff
@@ -17,7 +17,52 @@
 // Package stateconf configures state management.
 package stateconf
 
-import "github.com/ava-labs/libevm/libevm/options"
+import (
+	"github.com/ava-labs/libevm/libevm/options"
+)
+
+// A StateDBCommitOption configures the behaviour of all update calls within the
+// state.StateDB.Commit() implementations.
+// This is provided in two distinct types to allow customizability to both
+// snapshot and trie database updates, but are ignored in libevm implementations.
+type StateDBCommitOption = options.Option[stateDBCommitConfig]
+
+type stateDBCommitConfig struct {
+	snapshotOpts []SnapshotUpdateOption
+	triedbOpts   []TrieDBUpdateOption
+}
+
+// WithSnapshotUpdateOpts returns a StateDBCommitOption carrying a list of
+// SnapshotUpdateOptions.
+// If the list is not of length 1, the last option in the list is used.
```

Comment:
```suggestion
// If multiple such options are used, only the last will be applied as they overwrite each other.
```

The `...` makes `opts` a [variadic argument](https://gobyexample.com/variadic-functions), which allows passing of any number (including zero) of them to the function. Within the function body, however, it is treated as a `[]SnapshotUpdateOption`.

This pattern is less important in this particular function, and I only do it to avoid forcing users to make slices. Where it's extremely useful in `libevm` is in pre-existing geth signatures, exploiting the fact that zero arguments are allowed (i.e. the default, upstream behaviour).